src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java

```diff
@@ -269,45 +270,44 @@ else if (status.getOrderStatus().contains("CANCELLED") && (StringUtils.isBlank(k
         }
     }
 
-    public static void processingSingleConfirmation(Response gbfResponse, List<KitRequest> kitRequests, ShippingConfirmation confirmation) throws Exception {
-        //update external shipper response at request
+    public static void processingSingleConfirmation(Response gbfResponse, ShippingConfirmation confirmation) throws Exception {
         logger.info("Got confirmation for " + confirmation.getOrderNumber());
         Node node = GBFRequestUtil.getXMLNode(gbfResponse.getXML(), XML_NODE_EXPRESSION.replace("%1", confirmation.getOrderNumber()));
-        List<String> dsmKitRequestIds = getDSMKitRequestId(confirmation.getOrderNumber());
-        if (dsmKitRequestIds != null && !dsmKitRequestIds.isEmpty()) {
-            for (String dsmKitRequestId : dsmKitRequestIds) {
-                KitRequestExternal.updateKitRequestResponse(getStringFromNode(node), dsmKitRequestId);
-            }
-        }
-        //update kit shipping information
-        if (confirmation.getItem() != null) {
+        String externalResponse = getStringFromNode(node);
+        if (confirmation.getItem() != null && StringUtils.isNotBlank(externalResponse)) {
             Item item = confirmation.getItem();
-            List<KitRequest> kitRequestsResult = getKitRequest(kitRequests, confirmation.getOrderNumber(), item.getItemNumber(), 1);
-            int counter = 0;
-            if (kitRequestsResult != null) {
-                for (KitRequest kitRequest : kitRequestsResult) {
-                    String kitLabel = item.getTubeSerial();
-                    if (counter > 0) {
-                        kitLabel += "_" + counter;
+            inTransaction((conn) -> {
+                List<String> dsmKitRequestIds = getDSMKitRequestId(conn, confirmation.getOrderNumber()); //dsm_kit_request_id
```

Comment:
because we are getting here the list of dsm_kit_request_ids we don't need to give the list in from before

Reply:
we could maybe add a check, if the kit already has confirmation and all other informations, then don't update it again… but seeing gbf changing the xml I think we should leave it like that for now…

Reply:
I agree!